daemon/src/payout_curve.rs

```diff
@@ -721,13 +727,62 @@ mod tests {
         pretty_assertions::assert_eq!(actual_payouts, expected_payouts);
     }
 
+    #[test]
+    fn verify_effect_of_funding_fee() {
+        let payouts = calculate_payout_parameters(
+            Price::new(dec!(54000.00)).unwrap(),
+            Usd::new(dec!(3500.00)),
+            Leverage::new(5).unwrap(),
+            200,
+            0,
+        )
+        .unwrap();
+
+        let funding_fee = 100;
+        let payouts_with_fee = calculate_payout_parameters(
+            Price::new(dec!(54000.00)).unwrap(),
+            Usd::new(dec!(3500.00)),
+            Leverage::new(5).unwrap(),
+            200,
+            funding_fee,
+        )
+        .unwrap();
+
+        let sum_of_payouts_is_constant =
+            payouts
+                .iter()
+                .zip(payouts_with_fee.clone())
+                .all(|(payout, payout_with_fee)| {
+                    dbg!(payout.short_amount) + dbg!(payout.long_amount)
+                        == dbg!(payout_with_fee.short_amount) + dbg!(payout_with_fee.long_amount)
+                });
+
+        let short_gets_fee =
+            payouts
+                .iter()
+                .zip(payouts_with_fee.clone())
+                .all(|(payout, payout_with_fee)| {
+                    let total = payout_with_fee.short_amount + payout_with_fee.long_amount;
+                    let space_for_fee = total - payout_with_fee.short_amount;
+                    let expected_fee = funding_fee.min(space_for_fee);
+                    let fee = payout_with_fee.short_amount - payout.short_amount;
+
+                    fee == expected_fee
+                });
```

Comment:
```suggestion
        let actual_fees =
            payouts
                .iter()
                .zip(payouts_with_fee.clone())
                .map(|(payout, payout_with_fee)| payout_with_fee.short_amount - payout.short_amount);
                
        assert!(actual_fees.all(|fee| fee == 200));
```

Best to split the assertion out and assert against something static!

Reply:
This does look better but it won't pass for the payout which pays the short all of the money. I had to introduce the `expected_fee` to account for ranges which already pay the short position so much that there isn't enough money left from the total to cover the entire fee.

Reply:
> This does look better but it won't pass for the payout which pays the short all of the money.

We could treat it as a special case, but the same problem could happen for any payout which would originally pay the long position less than the funding fee.

I've been thinking about it and I think it's gonna be hard to have a nice

```rust
assert!(fees.all(|fee| fee == funding_fee));
```

like you propose.

Reply:
Can we assert against an array of fees then at least? Then we can add comments to the special cases?